src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -0,0 +1,345 @@
+
+            foreach (var property in values)
+            {
+                string stringValue = null;
+
+                // drop null properties
+                if (property.Value == null)
+                {
+                    continue;
+                }
+
+                // Format dates
+                Type propertyType = property.Value.GetType();
+                if (propertyType == typeof(DateTime))
+                {
+                    stringValue = ((DateTime)property.Value).ToUniversalTime().ToString(DateTimeFormatString);
+                }
+                else if (propertyType == typeof(DateTimeOffset))
+                {
+                    stringValue = ((DateTimeOffset)property.Value).UtcDateTime.ToString(DateTimeFormatString);
+                }
+                else
+                {
+                    stringValue = property.Value.ToString();
+                }
+
+                // Since there is no nesting of properties, apply a prefix before the property name to lessen
+                // the chance of collisions.
+                telemetry.Properties.Add(LoggingKeys.CustomPropertyPrefix + property.Key, stringValue);
+            }
+        }
+
+        private void LogFunctionResultAggregate(IEnumerable<KeyValuePair<string, object>> values)
+        {
+            // Metric names will be created like "{FunctionName} {MetricName}"
+            IDictionary<string, double> metrics = new Dictionary<string, double>();
+            string functionName = LoggingConstants.Unknown;
+
+            // build up the collection of metrics to send
+            foreach (KeyValuePair<string, object> value in values)
+            {
+                switch (value.Key)
+                {
+                    case LoggingKeys.Name:
+                        functionName = value.Value.ToString();
+                        break;
+                    case LoggingKeys.Timestamp:
+                    case LoggingKeys.OriginalFormat:
+                        // Timestamp is created automatically
+                        // We won't use the format string here
+                        break;
+                    default:
+                        if (value.Value is TimeSpan)
+                        {
+                            // if it's a TimeSpan, log the milliseconds
+                            metrics.Add(value.Key, ((TimeSpan)value.Value).TotalMilliseconds);
+                        }
+                        else if (value.Value is double || value.Value is int)
+                        {
+                            metrics.Add(value.Key, Convert.ToDouble(value.Value));
+                        }
+
+                        // do nothing otherwise
+                        break;
+                }
+            }
+
+            foreach (KeyValuePair<string, double> metric in metrics)
+            {
+                _telemetryClient.TrackMetric($"{functionName} {metric.Key}", metric.Value);
```

Comment:
I was about to incorporate it last week and then realized there may be drawbacks with building charts in the Portal (I've never seen it display these details before). I just reached out to Sergey about this, so let's see what he says.